File: 0005/LongestPalindromicSubstringFailedDueToTimeLimitExceeded.swift
class Solution {
    /**
     * Brute Force
     */
    func longestPalindrome(_ s: String) -> String {
        // Get length of input string
        let strArray = Array(s)
        let strLength = strArray.count

        // All substrings of length 1 are palindromic
        var maxLength = 1, startIndex = 0

        // Nested loop to mark start and end index
        for right in 0 ..< strLength {
            for left in right ..< strLength {
                var isFound = true

                // Check palindrome
                for offset in 0 ..< (left - right + 1) / 2 where strArray[right + offset] != strArray[left - offset] {
                    isFound = false
                }

                // Palindrome
                if isFound && (leftIndex - rightIndex + 1) > maxLength {
                    startIndex = rightIndex
                    maxLength = leftIndex - rightIndex + 1
                }
            }
        }

        return String(s.prefix(startIndex + maxLength).suffix(maxLength))
    }
}

File: 0005/LongestPalindromicSubstringFailedDueToTimeLimitExceeded.java
class Solution {
    /**
     * Brute Force
     */
    public String longestPalindrome(String s) {
        // Get length of input string
        int n = s.length();
        
        // All substrings of length 1 are palindromic
        int maxLength = 1, start = 0;
        
        // Nested loop to mark start and end index
        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j++) {
                boolean isFound = true;
                
                // Check paliindrome
                for (int k = 0; k < (j - i + 1) / 2; k++) {
                    if (s.charAt(i + k) != s.charAt(j - k)) {
                        isFound = false;
                    }
                }
                
                // Palindrome
                if (isFound && (j - i + 1) > maxLength) {
                    start = i;
                    maxLength = j - i + 1;
                }
            }
        }
        
        return s.substring(start, start + maxLength);
    }
}
